draft-thomson-httpbis-h2-stream-limits.md

```diff
@@ -53,18 +57,23 @@ of abuse.
 
 # Introduction
 
-HTTP/2 {{RFC9113}} allows endpoints to describe a concurrency limit for streams
-(`SETTINGS_MAX_CONCURRENT_STREAMS`).  Initially, the stream concurrency limit is
-only bounded by the maximum number of streams that can be created, which is
-2<sup>30</sup> for each endpoint, but the value can be changed at any time.
-Most endpoints set a smaller value in an attempt to protect the resources that
-are committed when processing a stream.
-
-This limit is not effective in the case that streams are quickly cancelled.
-Stream cancellation with the `RST_STREAM` frame has immediate effect, which
-means that the stream no longer counts against the concurrent stream limit.
-This means that a malicious endpoint can create and cancel an unbounded number
-of streams as long as its peer does not set a limit of zero.
+HTTP/2 allows endpoints to declare the concurrency limit for streams created by
+their peer (`SETTINGS_MAX_CONCURRENT_STREAMS`; see {{Section 5.1.2 of
+RFC9113}}). Initially, the stream concurrency limit is only bounded by the
+maximum number of streams that can be created, which is 2<sup>30</sup> for each
+endpoint, but the value can be changed at any time. Most endpoints set a smaller
+value in an attempt to protect the resources that are committed when processing
+a stream.
+
+This limit is not effective in the case that streams are quickly cancelled or
+closed. The creator of a stream can cancel it using the `RST_STREAM` frame.  The
+creator of a stream can also purposefully send frames that violate HTTP/2 rules,
+forcing a compliant receiver to close the stream. Either of these methods has
```

Comment:
How about this?

```suggestion
This limit is not effective in the case that streams are quickly cancelled.
The creator of a stream can cancel it using the `RST_STREAM` frame.  The
creator of a stream could also cause its peer to send `RST_STREAM` by
purposefully sending frames that violate HTTP/2 rules, unless errors also
cause the peer to close the connection. Either of these methods has
```

Reply:
So ... I have an implementation that responds with a 400 Bad Request + END_STREAM :smile:

Reply:
Did you open an issue on HTTP/2 suggesting that we permit that reaction?

Reply:
yeh nah, this is that other discussion we were having on the list. My interpretation of https://datatracker.ietf.org/doc/html/rfc9114#section-4.1.2-5, specifically

> For malformed requests, a server MAY send an HTTP response indicating the error prior to closing or resetting the stream.

is that its fine not to send a RST_STREAM if I'm going to close the stream any way

Reply:
I know.  It's just that your reaction is off the books for now, so we don't need to say that it's acceptable here.

Reply:
Fair